code-highlighter.html

```diff
@@ -223,40 +223,24 @@
       },
 
       _splitHunk: function (e) {
-        var path = this._getComposedPath(e)[0];
+        var path = e.target;
         var lineNumber = Number(path.getAttribute('index')) + 1;
-        if (path.localName === 'iron-icon' && lineNumber !== this._amountLines) {
+        if (path.localName === 'iron-icon') {
           var newElem = {};
-          var x = this.code.split('\n');
-          newElem.code = x.slice(lineNumber, this.code.length - 1).join('\n');
-          newElem.beforeStart = path.getAttribute('before');
-          newElem.afterStart = path.getAttribute('after');
-          newElem.index = this.index;
-          this.code = x.slice(0, lineNumber).join('\n');
-          this.fire('add-hunk', newElem);
-        }
-      },
-
-      // For browsers that support neither Event.path and Event.composedPath.
-      // from: https://stackoverflow.com/questions/36845515/mouseevent-path-equivalent-in-firefox-safari
-      _getComposedPath: function (e) {
-        var path = e.path || (e.composedPath && e.composedPath()) || [];
-        if (path.length === 0) {
-          var currentElem = e.target;
-          while (currentElem) {
-            path.push(currentElem);
-            currentElem = currentElem.parentElement;
+          if (lineNumber !== this._amountLines) {
+            var x = this.code.split('\n');
+            newElem.code = x.slice(lineNumber, this.code.length - 1).join('\n');
+            newElem.beforeStart = path.getAttribute('before');
+            newElem.afterStart = path.getAttribute('after');
+            newElem.index = this.index;
+            this.code = x.slice(0, lineNumber).join('\n');
```

Comment:
I think we want to alter the element's code in the parent element. IMO clicking a line should just fire an event and let the user of this element decide what to do with the event. This way, `code-highlighter` can be used in more scenario's.